THE-app/app/(tabs)/booksScreen.tsx

```diff
@@ -30,15 +31,105 @@ const GET_BOOKS = gql`
 
 function BooksScreen() {
   const { loading, error, data } = useQuery(GET_BOOKS);
+  const [filterText, setFilterText] = useState('');
+  const [selectedSubject, setSelectedSubject] = useState('');
+  const [selectedLevel, setSelectedLevel] = useState('');
+
+  let subjectOptions: JSX.Element[] = [];
+  let levelOptions: JSX.Element[] = [];
+
+  if (!loading && !error && data) {
+    // Extract subjects and levels from the data
+    const subjectsSet = new Set<string>();
+    const levelsSet = new Set<string>();
+
+    data.viewer.books.hits.forEach((book: Book) => {
+      book.subjects.forEach((subject) => subjectsSet.add(subject.name));
+      book.levels.forEach((level) => levelsSet.add(level.name));
+    });
+
+    // Convert sets to arrays and sort them
+    const subjectsArray = Array.from(subjectsSet).sort();
+    const levelsArray = Array.from(levelsSet).sort();
+
+    // Create options for subjects
+    subjectOptions = subjectsArray.map((subject) => (
+      <Select.Item key={subject} label={subject} value={subject} />
+    ));
+
+    // Create options for levels
+    levelOptions = levelsArray.map((level) => (
```

Comment:
Because this whole block is not memoized, it will be run again each time the screen is rendered (on each keypress).